test/_functions.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -e
+
+# Configure backup directory
+function configure_backup_directory_target() {
+    local t=$1
+	local seqno=$2
+	# where will we store
+	# create the backups directory
+	# clear the target
+	# replace SEQ if needed
+	t2=${t/SEQ/${seqno}}
+	mkdir -p ${BACKUP_DIRECTORY_BASE}/${seqno}/data
```

Comment:
```sh
chmod -R 0777 ${BACKUP_DIRECTORY_BASE}/${seqno}
```

The script runs as root, but the uid in the container is not. This is why it fails with the permissions error.

Reply:
What do you know, that fixed it!

It makes perfect sense. I think I missed it before because everything was running as root, and the issue with the overlap of the tests muddied the water a little too! Looking through your test suite again, I can see that I missed the fact that you were setting the permissions to 777 for the backup folder path when I refactored the functionality out into the `_functions.sh` file.

Reply:
see: https://github.com/deitch/mysql-backup/pull/56/commits/d00b2efabc2f2968691b12de9ec53b78855ff2f4 for the changes